scipy/optimize/optimize.py

```diff
@@ -490,10 +497,15 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
     if retall:
         allvecs = [sim[0]]
 
-    if maxiter is None:
+    if maxiter is None and maxfun is None:
         maxiter = N * 200
-    if maxfun is None:
         maxfun = N * 200
+    elif maxiter is not None and maxfun is not None:
+        pass
+    elif maxiter is not None and maxiter != np.inf:
+        maxfun = np.inf
+    elif maxfun is not None and maxfun != np.inf:
+        maxiter = np.inf
```

Comment:
The logic seems a bit wrong here:  for `maxiter=None, maxfev=np.inf`, this ends up running with `maxiter=None` which probably ends up with None vs. int comparison, which raises an exception.

Probably best to just write
```
if maxiter is None and maxfun is None:
    ...
elif maxiter is None:
    maxiter = np.inf
elif maxfun is None:
    maxfun = np.inf
```